code/game/objects/effects/aliens.dm

```diff
@@ -94,12 +94,16 @@
 		qdel(src)
 		return
 
-	for(var/mob/living/carbon/human/H in loc)
-		H.acid_spray_crossed(acid_damage, slow_amt)
+	SEND_SIGNAL(T, COMSIG_ATOM_ACIDSPRAY_ACT, src) //Signal the turf
+	for(var/H in T)
 
-	for(var/atom/A in loc) //Infrastructure for other interactions
+		var/atom/A = H
 		SEND_SIGNAL(A, COMSIG_ATOM_ACIDSPRAY_ACT, src)
 
+		if(ishuman(A))
+			var/mob/living/carbon/human/H2 = H
+			H2.acid_spray_crossed(acid_damage, slow_amt)
```

Comment:
this should use the signal

Reply:
This is basically how it was, I just made it less retarded.